### PR TITLE
US948127: Investigate "Worker did not fail liveness check when out of memory"

### PR DESCRIPTION
**Pull request by jonny-mccoubrey**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=948127

### CODE REVIEW

File: worker-test/src/main/config/worker.sh [hunk 1]
```diff
@@ -22,9 +22,18 @@ export CAF_CONFIG_ENABLE_SUBSTITUTOR=false
 
 export CAF_WORKER_DISABLE_ZERO_PROGRESS_REPORTING=true
 
+if [ "$CRASH_DUMP_ON_OUT_OF_MEMORY_ERROR" == "true" ]
```
Comment (jonny-mccoubrey) [inline worker-test/src/main/config/worker.sh:25]: @dermot-hardy what do you think about this in the `worker.sh` script?

File: worker-test/src/main/config/worker.sh [hunk 2]
```diff
@@ -22,9 +22,18 @@ export CAF_CONFIG_ENABLE_SUBSTITUTOR=false
...
+fi
+
+if [ "$HEAP_DUMP_ON_OUT_OF_MEMORY_ERROR" == "true" ]
+then
+  CAF_WORKER_JAVA_OPTS="${CAF_WORKER_JAVA_OPTS} -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=${DUMP_PATH}heap_dump_${HOSTNAME}.hprof"
+fi
+
 cd /maven
 exec java $CAF_WORKER_JAVA_OPTS \
-    -Dpolyglot.engine.WarnInterpreterOnly=false \
```
Comment (dermot-hardy) [inline worker-test/src/main/config/worker.sh:27]: Why are you removing this line?

Comment (jonny-mccoubrey) [reply worker-test/src/main/config/worker.sh:27]: That's a mistake, sorry

File: worker-test/src/main/config/worker.sh (hunk 1 above)
Comment (dermot-hardy) [reply worker-test/src/main/config/worker.sh:25]: It seems reasonable.

**Pull request closed without merging**